main.go

```diff
@@ -257,10 +258,17 @@ func main() { // nolint:gocyclo
 		setupLog.Error(err, "unable to create controller", "controller", "ToolchainConfig")
 		os.Exit(1)
 	}
+
+	githubClient, err := commonclient.NewGitHubClient(crtConfig.GitHubSecret().AccessTokenKey())
+	if err != nil {
+		setupLog.Error(err, "unable to create github client")
+		os.Exit(1)
+	}
```

Comment:
> we want to skip the check if no token is provided, but this will pretty much block start of the operator completely, right? Or what does happen when with initiating the client when the token is empty?

it's shouldn't be an issue, in the e2e test env actually we don't provide a token and it doesn't seem to cause any error/problem, of course it will not use this github client in the reconcile loops.

>  initiating the client in the main.go brings one complication - we need to restart the operator to be able to update the token (or to provide one if it hasn't been set already)

correct! but I did it this way , so I could unit test it. Not sure if there is any other way to test it and initialize the client inside the controller code. do you know ? Even using `gock` I think we still need to pass the github client as a parameter to the reconcile function right?

Reply:
there technically two ways:
1. initiate the client for every call separately. In that case you would pass a constructor function for the client, so you could provide different function for unit test.
2. introduce simple shared storage for the client - it would need to contain locks because it would be shared between:
    * ToolchainConfig so when the key is updated then it automatically regenerates the client
    * the ToolchainStatus that would use the client

Reply:
ok, thanks for your suggestions! I'll improve this part so we don't have to eventually restart the operator when there is a token update.

Reply:
I've implemented option 1) constructor function solution , please see https://github.com/codeready-toolchain/host-operator/pull/819/commits/d0384c77815b726c068230f89c018fdce41f3c5e